libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -295,10 +296,19 @@ export class ProjectChangeAnalyzer {
       const lookupByPath: IReadonlyLookupByPath<RushConfigurationProject> =
         this._rushConfiguration.getProjectLookupForRoot(rootDirectory);
 
+      let filterPath: string[] = [];
+
+      if (
+        projectSelection &&
+        this._rushConfiguration.experimentsConfiguration.configuration.enableSubpathScan
+      ) {
+        filterPath = Array.from(projectSelection).map(({ projectFolder }) => projectFolder);
```

Comment:
For a followup PR; this feature will 100% break the build cache unless we update this to `Array.from(Selection.expandAllDependencies(projectSelection), ({ projectFolder }) => projectFolder)`;

File hashes for dependencies are absolutely necessary when calculating build cache entry ids, unless the only selected phases don't depend on upstream projects at all.

Reply:
I suppose `projectSelection` already includes all the projects that need to be built?

Reply:
Suppose the current dependency relationships are as follows:
![image](https://github.com/user-attachments/assets/0cc092d5-4e10-47d1-acda-44a3addabcdb)
When running `rush build --to packageA`, the `projectSelection` will include all related packages (packageA to packageF)?

Reply:
With `--to`, projectSelection includes all the dependencies; with `--only`, it does not. This was addressed by #5045 by expanding the project selection when invoking `ProjectChangeAnalyzer`.

Reply:
https://developer.microsoft.com/json-schemas/rush/v5/experiments.schema.json

@octogonz Could you help deploy a new schema endpoint that includes the `enableSubpathScan` field?